site/en/configure/coverage.md

```diff
@@ -188,8 +188,61 @@ py_test(
     ],
 )
 ```
-<!-- TODO: Allow specifying a target for `PYTHON_COVERAGE`, instead of having to use `$(location)` -->
 
+If you are using a hermetic python toolchain, instead of adding an attribute to
+every `py_test` rule you can instead add the coverage tool to the toolchain
+configuration.
+
+Because the [pip_install][pip_install_rule] rule depends on the python
+toolchain, it cannot be used to fetch the `coverage` module.
+Instead, add in your `WORKSPACE` e.g.
+
+```starlark
+http_archive(
+    name = "coverage_linux_x86_64"",
+    build_file_content = """
+py_library(
+    name = "coverage",
+    srcs = ["coverage/__main__.py"],
+    data = glob(["coverage/*", "coverage/**/*.py"]),
+    visibility = ["//visibility:public"],
+)
+""",
+    sha256 = "84631e81dd053e8a0d4967cedab6db94345f1c36107c71698f746cb2636c63e3",
+    type = "zip",
+    urls = [
+        "https://files.pythonhosted.org/packages/74/0d/0f3c522312fd27c32e1abe2fb5c323b583a5c108daf2c26d6e8dfdd5a105/coverage-6.4.1-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl",
+    ],
+)
+```
+
+Then configure your python toolchain as e.g.
+
+```starlark
+py_runtime(
+    name = "py3_runtime_linux_x86_64",
+    coverage_tool = "@coverage_linux_x86_64//:coverage",
```

Comment:
There is probably an optimization to be made somewhere in here to only have the coverage dependency when coverage is enabled. The isCodeCoverage() check in the Java code prevents it from being included at runtime, but it still causes Bazel to do various target and config resolution steps.

Coverage being enabled can be detected with a config setting, e.g.

```
config_setting(
    name = "coverage_enabled",
    values = {"collect_code_coverage": "true"},
)

py_library(
    name = "coverage_if_enabled",
    deps = select({
        ":coverage_enabled": [
            ":coverage",
        ],
        "//conditions:default": [],
    }),
)
```

I'm not sure how that fits into toolchain registration; can selects be used within WORKSPACE?

In any case, I'm find with figuring this out in a separate PR.

Reply:
Declaration of the toolchain is in a BUILD file, not the WORKSPACE.  However, I am not sure either about how `select` gets evaluated in toolchain resolution.  As I commented elsewhere I think eventually this bit of the doc should be just baked into `python_register_toolchains` (in `rules_python`) so that most users don't have to care about it.